src/pages/Bridge/ActionPanel/BridgeActionPanel.tsx

```diff
@@ -47,19 +48,24 @@ export const BridgeActionPanel = ({
     isArbitrum,
     hasAmount
   } = useBridgeActionPanel()
+  const networkSwitcherPopoverOpen = useModalOpen(ApplicationModal.NETWORK_SWITCHER)
 
   const selectPanel = () => {
     // No wallet
     if (!account) {
-      return <ButtonPrimary onClick={toggleWalletSwitcherPopover}>Connect wallet</ButtonPrimary>
+      return (
+        <ButtonPrimary onClick={toggleWalletSwitcherPopover} disabled={networkSwitcherPopoverOpen}>
+          {networkSwitcherPopoverOpen ? 'Switch network' : 'Connect wallet'}
+        </ButtonPrimary>
+      )
     }
 
     // Change network
     if (!isNetworkConnected && step !== BridgeStep.Collect) {
       return (
         <ButtonPrimary
           onClick={() =>
-            fromNetworkChainId === ChainId.MAINNET ? selectEthereum() : selectNetwork(fromNetworkChainId)
+            fromNetworkChainId === ChainId.MAINNET ? selectNetwork(ChainId.MAINNET) : selectNetwork(fromNetworkChainId)
```

Comment:
Same as above, however this time selectNetwork needs to be wrapped in `useCallback` before passing to onClick